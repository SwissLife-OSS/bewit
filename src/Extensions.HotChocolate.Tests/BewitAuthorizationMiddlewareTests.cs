using System;
using System.Threading.Tasks;
using HotChocolate;
using HotChocolate.Execution;
using Snapshooter.Xunit;
using Xunit;

namespace Bewit.Extensions.HotChocolate.Tests
{
    public class BewitAuthorizationMiddlewareTests
    {
        [Fact]
        public async Task Query_WhenAuthorize_Success()
        {
            // arrange
            IServiceProvider serviceProvider = TestHelpers.CreateSchema();
            var payload = "foo@bar.gmail.com";
            var token = await TestHelpers.CreateToken(serviceProvider, payload);

            // act
            IExecutionResult result = await TestHelpers.ExecuteQuery(serviceProvider, token);

            // assert
            result.MatchSnapshot();
        }

        [Fact]
        public async Task Query_WhenWrongToken_Fail()
        {
            // arrange
            IServiceProvider serviceProvider = TestHelpers.CreateSchema();
            var token = await TestHelpers.CreateBadToken();

            // act
            IExecutionResult result = await TestHelpers.ExecuteQuery(serviceProvider, token);

            // assert
            result.MatchSnapshot(options =>
                options.IgnoreField("Errors[0].Exception.StackTraceString"));
        }

        [Fact]
        public async Task Query_WhenNotAuthorize_Fail()
        {
            // arrange
            IServiceProvider serviceProvider = TestHelpers.CreateSchema();

            // act
            IExecutionResult result = await TestHelpers.ExecuteQuery(serviceProvider);

            // assert
            result.MatchSnapshot();
        }
    }
}
