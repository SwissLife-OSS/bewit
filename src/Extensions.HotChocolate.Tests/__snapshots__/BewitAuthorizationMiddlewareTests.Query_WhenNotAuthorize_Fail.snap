{
  "Label": null,
  "Path": null,
  "Data": {
    "foo": null
  },
  "Errors": [
    {
      "Message": "NotAuthorized",
      "Code": null,
      "Path": {
        "Parent": null,
        "Depth": 0,
        "Name": "foo"
      },
      "Locations": [
        {
          "Line": 1,
          "Column": 3
        }
      ],
      "Extensions": null,
      "Exception": null
    }
  ],
  "Extensions": null,
  "ContextData": null,
  "HasNext": null
}
